### DISCUSSION TITLE
`projects: open recent`

### DISCUSSION THREAD

**Discussion by JosephTLyons** [zed-industries/zed#6653] [Show and tell]
Did you ever wish you could open a project as quickly as you `cmd-p` a file from within your editor?  Well, in Zed, you absolutely can.  The action can be summoned through the command palette under `projects: open recent`, or, simply by using `alt-cmd-o`.

![SCR-20230218-rqv](https://user-images.githubusercontent.com/19867440/219906353-54b78044-2fe2-4a2c-98b1-08094546d7cf.jpeg)

You can quickly filter down to what you want and launch the project immediately.

Pro Tip:  You can use `projects: open recent` to focus on a window that is already open.  If your project is open in a different [macOS Space](https://support.apple.com/guide/mac-help/work-in-multiple-spaces-mh14112/mac), running `projects: open recent` and selecting that project will bring that space and Zed window into focus.

**Comment by gldtn** [comment]
This is awesome! Would also be nice to be able to switch between projects on the same Zed instance instead of a new window.

**Reply by Moshyfawn** [reply]
> I think since [0.125.0](https://zed.dev/releases/preview/0.125.0) Preview cmd-click opens a project in a new window, otherwise the project is open in the same window.

https://github.com/zed-industries/zed/discussions/6653#discussioncomment-8767936

**Comment by krstp** [comment]
I just want to say project management in ZED is amazing. With everyday it holistically gets stronger. Thanks!

**Comment by chuikoffru** [comment]
I'd like to have an opportunity right click on icon in dock, and see recent projects as is in VSCode.

<img width="915" alt="Снимок экрана 2024-02-12 в 17 18 38" src="https://github.com/zed-industries/zed/assets/8284619/9e5f62c1-bcdc-481f-9dbc-07269f3a15e9">

<img width="915" alt="Снимок экрана 2024-02-12 в 17 18 50" src="https://github.com/zed-industries/zed/assets/8284619/895310a4-35e4-4389-aedd-26aca0ad9bfc">

**Reply by Moshyfawn** [reply]
This feature request is being tracked in #8416 and has a person looking into its implementation.

**Comment by diocletiann** [comment]
@JosephTLyons could we have an option to open the other project in the same window?

**Reply by Moshyfawn** [reply]
I think since [`0.125.0`](https://zed.dev/releases/preview/0.125.0) Preview `cmd-click` opens a project in a new window, otherwise the project is open in the same window.

**Comment by haunt98** [comment]
How do I force zed always open blank project? It keeps open my last project which is annoying.

**Reply by Moshyfawn** [reply]
This is not yet available in Zed. You can +👍 the #7694 feature request.

**Comment by CodeyPacker** [comment]
I love this feature. I'm probably in the minority with this request, but it'd be amazing if I could opt in to all it show all of my projects. I have like 20 repos for I jump through daily, so that would be a game changer for me.

**Comment by etiennelacoursiere** [comment]
Love this feature! I too switch between multiple projects everyday, would be super cool if you could set a project path in your config and switch between them with a project switcher like the recent projects one.

**Comment by Navidda** [comment]
This is great, but it does not seem to work when working with remote projects. Is it going to be implemented?

**Reply by bjackman** [reply]
It works for me, I guess you are experiencing a bug?

**Comment by bjackman** [comment]
In VSCode, if you already have the other project open in another window, then this command (well, its equivalent) will just focus that window. It seems that (with `{"create_new_window": true}`) Zed will open another instance of the same project. Any chance of updating that?

**Comment by italodr** [comment]
This feature is very useful. ~~Do you also plan to add this option as keybinding? In vscode/cursor, it's possible to navigate between open projects using "option + tab" (Mac). This would greatly help in moving quickly between open projects.~~
Sorry, I just saw that it already exists, under `alt+cmd+o`
